python/xmljson.py

```diff
@@ -0,0 +1,216 @@
+    --xmljson-red-mb=<n>     Reducer memory in Yarn (MB)  [default: 4096]
+    --xmljson-red-mb-hp=<n>  Reducer memory in JVM  (MB)  [default: 3584]
+    -f --force               If set, will overwrite old output
+    -d --debug               Print debug logging
+"""
+import logging
+import os.path
+import subprocess
+import sys
+
+import docopt
+import hdfs
+
+
+logger = logging.getLogger(__name__)
+
+
+ERROR_MSG = """
+        This job has been killed by user
+
+*************************************************
+*                                               *
+*     You SHOULD double check your cluster !    *
+*                                               *
+*   The application might still be running...   *
+*                                               *
+*************************************************
+"""
+
+
+class XMLJSONConverter(object):
+
+    def __init__(self,
+                 input_path,
+                 output_path,
+                 name_node,
+                 user,
+                 queue,
+                 jar,
+                 class_,
+                 reducers,
+                 timeout,
+                 mapper_mb,
+                 mapper_mb_heap,
+                 reducer_mb,
+                 reducer_mb_heap,
+                 force,
+                 debug):
+
+        self.input_path = input_path
+        self.output_path = output_path
+        self.name_node = name_node
+        self.user = user
+        self.queue = queue
+        self.jar = jar
+        self.class_ = class_
+        self.reducers = str(int(reducers))
+        self.timeout = str(int(timeout) * 1000)
+        self.mapper_mb = str(int(mapper_mb))
+        self.mapper_mb_heap = str(int(mapper_mb_heap))
+        self.reducer_mb = str(int(reducer_mb))
+        self.reducer_mb_heap = str(int(reducer_mb_heap))
+        self.force = force
+        self.debug = debug
+
+        self._init_logging()
+
+    def _init_logging(self):
+        logging.basicConfig(
+            format='%(asctime)s %(levelname)s:%(name)s -- %(message)s'
+        )
+        logger.setLevel(logging.DEBUG if self.debug else logging.INFO)
+
+    def run(self):
+        self._configure_hdfs_client()
+        if self._prepare_hdfs():
+            try:
+                self._execute_hadoop_job()
+            except KeyboardInterrupt:
+                logger.error(ERROR_MSG)
+                raise RuntimeError("Job interrupted")
+            if not self._check_success_file():
+                raise RuntimeError("No success file after hadoop job, " +
+                                   "something probably went wrong")
+
+    def _configure_hdfs_client(self):
+        name_node = self.name_node
+        user = self.user
+        self.hdfs_client = hdfs.client.InsecureClient(name_node, user=user)
+
+    def _prepare_hdfs(self):
+        logger.debug("Preparing HDFS for xmljson")
+        if self._check_success_file():
```

Comment:
Is there a reason not to move `if self._check_success_file():` up into `run()`? I think this would help with code clarity, so it reads as "_Check if we really need to do something. Prepare to do it. Now actually execute it_" rather than, "_Prepare to do something, wait we didn't have to do anything, I guess we are done_"